### PR TITLE
feat: update analytics to new api usage

### PR DESCRIPTION
**Pull request by abutterworth**
exports `configure`, `SegmentAnalyticsService`, and all the existing analytics functions. Remove's an `initialize` function which is now executed internally upon calling `configure(impl, options)`.

New Usage:
```javascript 
configure(SegmentAnalyticsService, {
    loggingService,
    httpClient,
    analyticsApiBaseUrl,
    trackingLogApiBaseUrl: `${analyticsApiBaseUrl}/event`,
    apiKey: configuration.SEGMENT_KEY,
});
```

### CODE REVIEW

File: src/analytics/README.rst [hunk 1]
```diff
@@ -15,19 +15,20 @@ To install frontend-analytics into your project::
...
     // Sample object containing app configuration
     import { configuration } from '../config/environment';
 
-    initializeSegment(configuration.SEGMENT_KEY);
-    configureAnalytics({
+    configureAnalytics(SegmentAnalyticsService, {
       loggingService: LoggingService,
-      authApiClient: apiClient,
+      httpClient: apiClient,
       analyticsApiBaseUrl: configuration.LMS_BASE_URL,
```
Comment (davidjoy) [inline src/analytics/README.rst]: These config-based things can be snagged from the brand new ConfigService!  

```
import { getConfigService } from '../config';
import { getAuthenticatedHttpClient } from '../auth';
import { getLoggingService } from '../logging';


configureAnalytics(SegmentAnalyticsService, {
  configService: getConfigService(),
  httpClient: getAuthenticatedHttpClient()
  loggingService: getLoggingService(),
});
```

Something like that.

File: src/analytics/README.rst [hunk 2]
```diff
@@ -15,19 +15,20 @@ To install frontend-analytics into your project::
 
 Add the following configuration code to your app::
 
-    import { configureAnalytics, initializeSegment } from '@edx/analytics';
+    import { configure as configureAnalytics, SegmentAnalyticsService } from '@edx/analytics';
```
Comment (davidjoy) [inline src/analytics/README.rst]: The `@edx/analytics`, `@edx/frontend-logging`, etc, packages should all be updated here, presumably?

File: src/analytics/README.rst (hunk 1 above)
Comment (abutterworth) [reply src/analytics/README.rst]: 👍

File: src/analytics/README.rst (hunk 2 above)
Comment (abutterworth) [reply src/analytics/README.rst]: 👍

**Pull request merged**